pyuvdata/miriad.py

```diff
@@ -186,6 +200,62 @@ def read_miriad(self, filepath, correct_lat_lon=True, run_check=True,
         for extra_variable in extra_miriad_variables:
             check_variables[extra_variable] = uv[extra_variable]
 
+        # select on ant_str if provided
+        if ant_str is not None:
+            # typc check
+            assert isinstance(ant_str, (str, np.str)), "ant_str must be fed as a string"
+            uv.select(ant_str, 0, 0)
```

Comment:
I think this line only works if 'auto' is passed. 'cross' and 'all' do not work (at least with my aipy). It might be better to use scripting.uv_selector instead.